utils/R/convert.input.R

```diff
@@ -6,65 +6,144 @@
+            remote.execute.R(file.deletion.commands$delete.tmp, host, user=NA, verbose=TRUE, R="R")
+          } else {
+            logger.info("Conversion failed. Replacing old files.")
+            remote.execute.R(file.deletion.commands$replace.from.tmp, host, user=NA, verbose=TRUE, R="R")
+          }
+        )
+      }
+    } else if((start_date >= existing.input$start_date) && (end_date <= existing.input$end_date)) {
+      # There's an existing input that spans desired start/end dates. Use that one. 
+      logger.info("Skipping this input conversion because files are already available.")
+      return(list(input.id=existing.input$id, dbfile.id=existing.dbfile$id))
+    } else {
+      # Start/end dates need to be updated so that the input spans a continuous timeframe
+      start_date <- min(start_date, existing.input$start_date)
+      end_date <- max(end_date, existing.input$end_date)
+      logger.info(paste0("Changed start/end dates to '", start_date, "'/'", end_date, "' ",
+                  "so that existing input can be updated while maintaining continuous time span."))
+      
+      # There might be existing files for some years (but not all; checked that above)
+      # fcn should be smart enough not overwrite the existing ones, 
+      # and hopefully won't waste time working on them either
+      # At the end, if convert.inputs was successful we'll need to update its start/end dates
+      # We don't know the dbfile path/prefix until after fcn runs, so unfortunately can't check 
+      # that the new dbfile record won't conflict with existing ones. 
+    }
+  } else {
+    # No existing record found. Should be good to go?
   }
   
-  input = db.query(paste("SELECT * from inputs where id =",input.id),con)
-  if(nrow(input)==0){logger.error("input not found",input.id);return(NULL)}
   
-  ifelse(host$name == "localhost", machine.host <- fqdn(), machine.host <- host$name)
-  machine = db.query(paste0("SELECT * from machines where hostname = '",machine.host,"'"),con)
-  # machine = db.query(paste("SELECT * from machines where id = ",dbfile$machine_id),con)
-  if(nrow(machine)==0){logger.error("machine not found",host$name);return(NULL)}
   
-  # dbfile may return more than one row -> may need to loop over machine ids
-  dbfile = db.query(paste("SELECT * from dbfiles where container_id =",input.id," and container_type = 'Input' and machine_id =",machine$id),con)
-  if(nrow(dbfile)==0){logger.error("dbfile not found",input.id);return(NULL)}
-  if(nrow(dbfile)>1){
-    logger.warning("multiple dbfile records, using last",dbfile);
-    dbfile = dbfile[nrow(dbfile),]
+  machine.host <- ifelse(host$name == "localhost", fqdn(), host$name)
+  machine = db.query(paste0("SELECT * from machines where hostname = '", machine.host, "'"), con)
+  
+  if(nrow(machine)==0){
+    logger.error("machine not found", host$name)
+    return(NULL)
   }
 
+  if(missing(input.id) || is.na(input.id) || is.null(input.id)) {
```

Comment:
Here's one of just a few places where it actually matters whether convert.inputs was passed an input.id (as mentioned, download.raw.met now uses this too, but there's no input for that one of course).